File: scala/b03.scala

import  scala.io.StdIn._

object Main {
    def main(args: Array[String]): Unit= {
        val n = readInt()
        val listA = readLine().split(" ").map(_.toInt)
        val isFound = (for {
                        i <- listA
                        j <- listA if j > i
                        k <- listA if k > j
                    } yield i + j + k ).filter(_ == 1000).length
        val ans = if (isFound > 0) "Yes" else "No"
        println(ans)
    }
    
}

File: cpp/b03.cpp
#include <iostream>
using namespace std;

int main() {
    int N;
    int A[109];
    cin >> N;
    for (int i = 0; i < N; i++) cin >> A[i];
    for (int i = 0; i < N; i++) {
        for (int j = i+1; j < N; j++) {
            for (int k = j + 1; k < N; k++) {
                if (A[i] + A[j] + A[k] == 1000 ) {
                    cout << "Yes";
                    return 0;
                }
            }
        }
    }
    cout << "No";
    return 0;
}